prime-router/src/test/kotlin/serializers/Hl7SerializerTests.kt

```diff
@@ -158,4 +161,78 @@ NTE|1|L|This is a final comment|RE"""
+        every { terser.get("/.${element.hl7Field}-5") } returns ""
+        every { terser.get("/.${element.hl7Field}-6") } returns ""
+        every { terser.get("/.${element.hl7Field}-7") } returns ""
+        phoneNumber = serializer.decodeXTNPhoneNumber(terser, element)
+        assertEquals("5555555555", phoneNumber)
+
+        // Phone number in newer components.  Will ignore phone number in deprecated component
+        every { terser.get("/.${element.hl7Field}-1") } returns "4444444444"
+        every { terser.get("/.${element.hl7Field}-2") } returns ""
+        every { terser.get("/.${element.hl7Field}-3") } returns "PH" //Phone
+        every { terser.get("/.${element.hl7Field}-4") } returns ""
+        every { terser.get("/.${element.hl7Field}-5") } returns "555"
+        every { terser.get("/.${element.hl7Field}-6") } returns "666"
+        every { terser.get("/.${element.hl7Field}-7") } returns "7777"
+        phoneNumber = serializer.decodeXTNPhoneNumber(terser, element)
+        assertEquals("5556667777", phoneNumber)
+
+        // No type assumed to be a phone number
+        every { terser.get("/.${element.hl7Field}-1") } returns ""
+        every { terser.get("/.${element.hl7Field}-2") } returns ""
+        every { terser.get("/.${element.hl7Field}-3") } returns ""
+        every { terser.get("/.${element.hl7Field}-4") } returns ""
+        every { terser.get("/.${element.hl7Field}-5") } returns "555"
+        every { terser.get("/.${element.hl7Field}-6") } returns "666"
+        every { terser.get("/.${element.hl7Field}-7") } returns "7777"
+        phoneNumber = serializer.decodeXTNPhoneNumber(terser, element)
+        assertEquals("5556667777", phoneNumber)
+
+        // A Fax number is not used
+        every { terser.get("/.${element.hl7Field}-1") } returns ""
+        every { terser.get("/.${element.hl7Field}-2") } returns ""
+        every { terser.get("/.${element.hl7Field}-3") } returns "FX" //Fax
+        every { terser.get("/.${element.hl7Field}-4") } returns ""
+        every { terser.get("/.${element.hl7Field}-5") } returns "555"
+        every { terser.get("/.${element.hl7Field}-6") } returns "666"
+        every { terser.get("/.${element.hl7Field}-7") } returns "7777"
+        phoneNumber = serializer.decodeXTNPhoneNumber(terser, element)
+        assertEquals("", phoneNumber)
+
+        // Test null handling
+        every { terser.get("/.${element.hl7Field}-1") } returns null
+        every { terser.get("/.${element.hl7Field}-2") } returns null
+        every { terser.get("/.${element.hl7Field}-3") } returns null
+        every { terser.get("/.${element.hl7Field}-4") } returns null
+        every { terser.get("/.${element.hl7Field}-5") } returns null
+        every { terser.get("/.${element.hl7Field}-6") } returns null
+        every { terser.get("/.${element.hl7Field}-7") } returns null
+        phoneNumber = serializer.decodeXTNPhoneNumber(terser, element)
+        assertEquals("", phoneNumber)
+    }
```

Comment:
Please build a test for a field with repetition in the email field.

It would look like this:

`^NET^Internet^ouida.sanford@email.com~(250)1801593^PRN^PH^^1^250^1801593`

Reply:
Arg!  I followed too much the example data and forgot to check the spec.  Some things will have to change.  Will work on changes.  Nice catch!

Reply:
Added the test with the email address, but any non-PH type is ignored.